docs/tutorials/forecasting/howto_pytorch_lightning.md

```diff
@@ -0,0 +1,346 @@
+
+        distr_args, loc, scale = self(context)
+        distr = self.distr_output.distribution(distr_args, loc, scale)
+        loss = -distr.log_prob(target)
+
+        return loss.mean()
+
+    def configure_optimizers(self):
+        optimizer = torch.optim.Adam(self.parameters(), lr=1e-3)
+        return optimizer
+```
+
+We can now instantiate the training network, and explore its set of parameters.
+
+
+```python
+freq = "1H"
+context_length = 2 * 7 * 24
+prediction_length = dataset.metadata.prediction_length
+hidden_dimensions = [96, 48]
+```
+
+
+```python
+net = LightningFeedForwardNetwork(
+    freq=freq,
+    prediction_length=prediction_length,
+    context_length=context_length,
+    hidden_dimensions=hidden_dimensions,
+    distr_output=StudentTOutput(),
+)
+```
+
+
+```python
+sum(np.prod(p.shape) for p in net.parameters())
+```
+
+
+```python
+for p in net.parameters():
+    print(p.shape)
+```
+
+## Defining the training data loader
+
+We now set up the data loader which will yield batches of data to train on. Starting from the original dataset, the data loader is configured to apply the following transformation, which does essentially two things:
+* Replaces `nan`s in the target field with a dummy value (zero), and adds a field indicating which values were actually observed vs imputed this way.
+* Slices out training instances of a fixed length randomly from the given dataset; these will be stacked into batches by the data loader itself.
+
+
+```python
+from gluonts.dataset.field_names import FieldName
+from gluonts.transform import AddObservedValuesIndicator, InstanceSplitter, ExpectedNumInstanceSampler, TestSplitSampler
+```
+
+
+```python
+mask_unobserved = AddObservedValuesIndicator(
+    target_field=FieldName.TARGET,
+    output_field=FieldName.OBSERVED_VALUES,
+)
+```
+
+
+```python
+training_splitter = InstanceSplitter(
+    target_field=FieldName.TARGET,
+    is_pad_field=FieldName.IS_PAD,
+    start_field=FieldName.START,
+    forecast_start_field=FieldName.FORECAST_START,
+    instance_sampler=ExpectedNumInstanceSampler(
+        num_instances=1,
+        min_future=prediction_length,
+    ),
+    past_length=context_length,
+    future_length=prediction_length,
+    time_series_fields=[FieldName.OBSERVED_VALUES],
+)
+```
+
+
+```python
+from gluonts.dataset.loader import TrainDataLoader
+from gluonts.itertools import Cached
+from gluonts.torch.batchify import batchify
+```
+
+
+```python
+batch_size = 32
+num_batches_per_epoch = 50
+```
+
+
+```python
+data_loader = TrainDataLoader(
+    # We cache the dataset, to make training faster
+    Cached(dataset.train),
+    batch_size=batch_size,
+    stack_fn=batchify,
+    transform=mask_unobserved + training_splitter,
+    num_batches_per_epoch=num_batches_per_epoch,
+)
+```
+
+## Train the model
+
+We can now train the model using the tooling that PyTorch Lightning provides:
+
+
+```python
+trainer = pl.Trainer(max_epochs=10)
```

Comment:
`trainer = pl.Trainer(max_epochs=10, gpus=-1)` will also make it work on gpus

Reply:
doesn't that depend on how the data loader is set up? which device to load batches to

Reply:
lightning will move the batches to the appropriate device..

Reply:
actually this will work: `pl.Trainer(max_epochs=10, gpus=-1 if torch.cuda.is_available() else None)`